Src/FluentAssertions/NumericAssertionsExtensions.cs

```diff
@@ -1481,6 +1481,209 @@ public static class NumericAssertionsExtensions
 
     #endregion
 
+    #region BeNaN
+
+    /// <summary>
+    /// Asserts that the float value is NaN.
```

Comment:
🤔 I prefer to keep this more functional, e.g. like `Asserts that a number is seen as not-a-number`.